### ISSUE TITLE
agent includes entries that later cause native-image build failure: AggregatedMemoryPoolBean

### ISSUE THREAD

**Issue by aclement** [oracle/graal#2988]
**Describe the issue**
I am using the agent to collect the configuration data for an application. It produces a reflect-config.json containing org.graalvm types:
```
{
  "name":"org.graalvm.compiler.hotspot.management.AggregatedMemoryPoolBean",
  "allPublicConstructors":true
},
{
  "name":"org.graalvm.nativeimage.ImageInfo",
  "methods":[{"name":"inImageCode","parameterTypes":[] }]
},
```
however, on then building with native-image and supplying that config at least the first entry causes:
```
Error: type is not available in this platform: org.graalvm.compiler.hotspot.management.AggregatedMemoryPoolBean
Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
Error: Image build request failed with exit status 1
com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
	at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1647)
	at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1397)
	at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1358)
	at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1317)
	at com.oracle.svm.driver.NativeImage$JDK9Plus.main(NativeImage.java:1829)
```

**Steps to reproduce the issue**
Please include both build steps as well as run steps
I am using graalvm-ce-java11-20.3.0-dev-20201112
1. `git clone https://github.com/spring-projects-experimental/spring-graalvm-native.git`
2. `cd spring-graalvm-native/spring-graalvm-native-samples/actuator-webmvc`
3. `mvn clean package`
4. `mkdir -p graal/META-INF/native-image`
5. `java -agentlib:native-image-agent=config-output-dir=graal/META-INF/native-image -jar target/actuator-webmvc-0.0.1-SNAPSHOT.jar`
6. Ctrl-C to quit the app
7. Review `graal/META-INF/native-image/reflect-config.json` see the AggregatedMemoryPoolBean
8. If you want to build it then open `compile.sh`, on line 33, change `-cp $CP $MAINCLASS >> output.txt ; } 2>> output.txt` to `-cp $CP:../../graal $MAINCLASS >> output.txt ; } 2>> output.txt`
9. Run `compile.sh`
10. See error:
```
Error: type is not available in this platform: org.graalvm.compiler.hotspot.management.AggregatedMemoryPoolBean
Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
Error: Image build request failed with exit status 1
com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
	at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1647)
	at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1397)
	at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1358)
	at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1317)
	at com.oracle.svm.driver.NativeImage$JDK9Plus.main(NativeImage.java:1829)
```

**Describe GraalVM and your environment:**
 - GraalVM version graalvm-ce-java11-20.3.0-dev-20201112
 - JDK major version: **11]**
 - OS: **macOS Catalina**

**More details**
```
native-image --version
GraalVM Version 20.3.0-dev (Java Version 11.0.9+10-jvmci-20.3-b06)
```

**Comment by dougxc** [comment]
@tzezula can you please look into this.

**Comment by vjovanov** [comment]
@peter-hofer we have the following filter in the `AccessAdvisor`: 
```
   internalCallerFilter.addOrGetChildren("org.graalvm.compiler.**", RuleNode.Inclusion.Exclude);
```

Shouldn't this filter out the entry in question?

**Comment by peter-hofer** [comment]
> @peter-hofer we have the following filter in the `AccessAdvisor`:
> 
> ```
>    internalCallerFilter.addOrGetChildren("org.graalvm.compiler.**", RuleNode.Inclusion.Exclude);
> ```
> 
> Shouldn't this filter out the entry in question?

@vjovanov Not the entry you reference, which applies to lookups from _callers_ in `org.graalvm.compiler.**`. However, we also have a filter for lookups of elements in `org.graalvm.compiler.**`, which should cover this, so this could be a bug:
```
accessWithoutCallerFilter.addOrGetChildren("org.graalvm.compiler.**", RuleNode.Inclusion.Exclude);
```

**Comment by peter-hofer** [comment]
On second thought, `accessWithoutCallerFilter` applies only when the agent cannot determine the caller. Judging from the context, `AggregatedMemoryPoolBean` seems to be accessed from the management bean mechanism, which is probably not covered by a caller filter. The question is if it should be because it can also be used to load custom management beans.

**Comment by vjovanov** [comment]
I think it would make sense to apply a direct filter even when we have a caller. Sometimes it is just easier to have a class excluded irrespective of the caller. This would be useful whenever the caller is used for both JDK and user classes.

**Comment by cyw3** [comment]
I have the same error.

**Comment by mostafacs** [comment]
I have the same error with quarkus and kafka.

**Comment by peter-hofer** [comment]
Thanks everyone for the feedback.

> I think it would make sense to apply a direct filter even when we have a caller. Sometimes it is just easier to have a class excluded irrespective of the caller. This would be useful whenever the caller is used for both JDK and user classes.

I agree. We should go further and filter anything in GraalVM that we don't export (in terms of the Java module system) to application code. @vjovanov, could the tooling you used for class initialization list be adapted for that?

**Comment by helpermethod** [comment]
For anyone looking for a temporary workaround:

Just add the following entry to `access-filter.json`

```
    {
      "excludeClasses": "org.graalvm.compiler.hotspot.management.AggregatedMemoryPoolBean"
    }
```

**Comment by peter-hofer** [comment]
Fixed in 6c68fa3253cbeae04265f99f73631d725c335300.
